cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/util/ExtractTemplatesUtil.java

```diff
@@ -45,7 +46,17 @@ public static void extractTemplates(Path extractTo, boolean forceOverride) throw
     // find templates will also download jars if needed as a side effect and will return the path to the
     // files.
     URI findTemplatesLocation = ConfigurationFinder.findTemplatesLocation();
-    if (Files.isDirectory(Paths.get(findTemplatesLocation))) {
+
+    Path oldCobiGenTemplatesLocation = Paths.get(findTemplatesLocation);
+
+    if (Files.exists(oldCobiGenTemplatesLocation)) {
```

Comment:
`findTemplatesLocation` can also return the template-sets folder, right? So don't you need to check in this condition if the returned location is the belongs to the old structure?